hack/sync-from-kubernetes.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+
+dir=$(mktemp -d "${TMPDIR:-/tmp/}$(basename 0).XXXXXXXXXXXX")
+echo ${dir}
+
+git remote add upstream-kube git@github.com:kubernetes/kubernetes.git  || true
+git fetch upstream-kube
+
+currBranch=$(git rev-parse --abbrev-ref HEAD)
+previousKubeSHA=$(cat kubernetes-sha)
+
+git branch -D kube-sync || true
+git checkout upstream-kube/master -b kube-sync
+git reset --hard upstream-kube/master
+newKubeSHA=$(git log --oneline --format='%H' kube-sync -1)
+git filter-branch -f --subdirectory-filter staging/src/k8s.io/apimachinery HEAD
```

Comment:
I'm usually not a fan of comments that say what code is doing but I think it might be instructive if somewhere this file there were a comment that gave the rough outline of what is going on.